jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -1322,13 +1394,15 @@ def _curve_of_growth(data, centroid, aperture, final_sum, wcs=None, background=0
     if pixarea_fac is not None:
         sum_arr = sum_arr * pixarea_fac
     if isinstance(final_sum, u.Quantity):
-        final_sum = final_sum.to(sum_arr.unit, equivalencies)
+        final_sum = flux_conversion_general(final_sum.value, final_sum.unit,
+                                            sum_arr.unit, equivalencies)
     sum_arr = np.append(sum_arr, final_sum)
 
     if sum_unit is None:
         y_label = 'Value'
     else:
         y_label = sum_unit.to_string()
-        sum_arr = sum_arr.to_value(sum_unit, equivalencies)  # bqplot does not like Quantity
+        sum_arr = flux_conversion_general(sum_arr.value, sum_arr.unit, sum_unit,

```

Comment:
I think the comment still applies and should be kept.
```suggestion
        # bqplot does not like Quantity
        sum_arr = flux_conversion_general(sum_arr.value, sum_arr.unit, sum_unit,
```